modules/dataquery/ajax/datadictionary.php

```diff
@@ -19,16 +19,34 @@
 $client = new NDB_Client();
 $client->makeCommandLine();
 $client->initialize(__DIR__ . "/../../../project/config.xml");
-$cdb      = CouchDB::singleton();
-$category = $_REQUEST['category'];
-$results  = $cdb->queryView(
-    "DQG-2.0",
-    "datadictionary",
-    array(
-     "reduce"   => "false",
-     "startkey" => "[\"$category\"]",
-     "endkey"   => "[\"$category\", \"ZZZZZZZZ\"]",
-    )
-);
+
+$cdb = CouchDB::singleton();
+if ($_REQUEST['category']) {
+    $category = $_REQUEST['category'];
+
+    $results = $cdb->queryView(
+        "DQG-2.0",
+        "datadictionary",
+        array(
+         "reduce"   => "false",
+         "startkey" => "[\"$category\"]",
+         "endkey"   => "[\"$category\", \"ZZZZZZZZ\"]",
+        )
+    );
+} else if ($_REQUEST['key']) {
+    $key = explode(',', $_REQUEST['key']);
+    // error_log($key);
+
+    $results = $cdb->queryView(
+        "DQG-2.0",
+        "datadictionary",
+        array(
+         "reduce" => "false",
+         "key"    => "[\"$key[0]\",\"$key[1]\"]",
```

Comment:
I don't think this is any worse than before, but shouldn't these keys be `urlencode()`ed? Or does queryView do that already? (same with the $category variable above..)